### ISSUE TITLE
Different handling of whitspace by Clojure lexer

### ISSUE THREAD

**Issue by rsslldnphy** [rouge-ruby/rouge#103]
Hi, I've been trying to investigate [this issue](https://github.com/kytrinyx/exercism.io/issues/821#issuecomment-25751515) with clojure syntax highlighting on [exercism.io](http://exercism.io), and it seems to be being caused by some weirdness with how `Loofah` sanitization deals with `span` elements without whitespace between them. If there are no spaces between the spans then it adds a newline between each one.

What does this have to do with Rouge? Well the problem only happens with Clojure (that I've experienced so far), and having a look at the generated html and the Rouge codebase it appears that the Clojure lexer deals with whitespace in a slightly different way to most of the other lexers, by adding a `<span class="w">` in place of the whitespace rather than just leaving it intact.

So basically I'm just wondering what the reason behind this difference is? At this stage I'm not really sure how to resolve the issue for exercism.io, or whether it's going to need a pull-request to Exercism, Rouge, Loofah or even Nokogiri (as that's what's used by Loofah under the hood).
